File: 65-hard-strategy-portfolio/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type input struct {
	N        int      `json:"n"`
	Pnl      []int64  `json:"pnl"`
	Requires [][2]int `json:"requires"`
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	fmt.Println(solve(in.N, in.Pnl, in.Requires))
}

File: 65-hard-strategy-portfolio/rust/src/main.rs
use std::io::{self, Read};
use strategy_portfolio::solve;
use strategy_portfolio::Input;

fn main() {
    let mut buf = String::new();
    io::stdin().read_to_string(&mut buf).unwrap();
    let inp: Input = serde_json::from_str(&buf).unwrap();
    println!("{}", solve(inp.n, &inp.pnl, &inp.requires));
}
